addons/web/models/ir_http.py

```diff
@@ -4,14 +4,22 @@
 import json
 
 from odoo import models
-from odoo.http import request
+from odoo.http import ALLOWED_DEBUG_MODES, request
 
 import odoo
 
 
 class Http(models.AbstractModel):
     _inherit = 'ir.http'
 
+    @classmethod
+    def _dispatch(self):
+        # Store URL debug mode (might be empty) into session
+        if 'debug' in request.httprequest.args:
+            debug_mode = request.httprequest.args.get('debug')
+            request.session.debug = debug_mode if debug_mode in ALLOWED_DEBUG_MODES else True
```

Comment:
don't alter the session unless you absolutely need it, ie. the debug mode has effectively changed

Reply:
Good catch, the if condition is now a bit more verbose but done :)

Reply:
Why here ?

Reply:
I moved it from `http_routing` to `web` as you can see on this [push force](https://github.com/odoo/odoo/compare/d3ace26a502a8515d3114f5bb69a2e05fba4d68d..d2c19fcf7b73374137d0eea0677ef1717a60bbc5) as this behavior is required even for non http_routing apps, else it would be inconsistent depending if the database as http_routing installed.